lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -96,22 +134,45 @@ export default function SubmitGradeForm({
   // Used to handle keyboard input changes for the grade input field.
   const inputRef = useRef<HTMLInputElement | null>(null);
 
-  // This is used to track an unsaved grade. It is null until user input occurs.
-  const [draftGradeValue, setDraftGradeValue] = useState<string | null>(null);
+  // This is used to track unsaved grades or comments. It is null until user input occurs.
+  const [draftGrading, setDraftGrading] = useState<DraftGrading>({
+    grade: null,
+    comment: null,
+  });
+  const updateDraftGrading = useCallback(
+    (partialDraftGrading: Partial<DraftGrading>) => {
+      const newDraftGrading = {
+        grade: partialDraftGrading.grade ?? draftGrading.grade,
+        comment: partialDraftGrading.comment ?? draftGrading.comment,
+      };
+
+      onUnsavedChanges?.(hasGradeChanged(newDraftGrading, grade.data));
+      setDraftGrading(prev => ({ ...prev, ...partialDraftGrading }));
+    },
+    [draftGrading.grade, draftGrading.comment, onUnsavedChanges, grade.data]
+  );
```

Comment:
Having individual peces of state for the grade and the comment made it a bit more difficult to determine if there are unsaved changes.

A common piece of state allows for this logic to be wrapped here, and invoke `onUnsavedChanges` at the same time we update the draft state.